packages/api/src/webhooks/cannon.ts

```diff
@@ -474,26 +487,31 @@ export default class WebhookCannon {
     event: messages.WebhookEvent,
     resp: Response,
     startTime: [number, number],
-    responseBody: string
+    responseBody: string,
+    stream: DBStream,
+    params
   ) {
     try {
       const hrDuration = process.hrtime(startTime);
-      let encodedResponseBody = Buffer.from(responseBody).toString("base64");
+      const encodedResponseBody = Buffer.from(
+        responseBody.substring(0, 1024)
+      ).toString("base64");
 
       await this.db.webhookResponse.create({
         id: uuid(),
         webhookId: webhook.id,
         eventId: event.id,
+        event: event.event,
+        userId: webhook.userId,
         createdAt: Date.now(),
         duration: hrDuration[0] + hrDuration[1] / 1e9,
-        statusCode: resp.status,
         response: {
           body: encodedResponseBody,
-          headers: resp.headers.raw(),
           redirected: resp.redirected,
           status: resp.status,
           statusText: resp.statusText,
         },
+        request: params,
```

Comment:
Can we have an explicit schema here? This params is a raw `fetch` parameters object and I think it could change inadvertently (or have weird stuff like functions in it). Even if we end up just copying specific fields here like url, header, etc, I think it'd still be preferrable. WDYT?

Reply:
I did that initially but then thought we needed everything for the resending, but being explicit is safer, i'll switch it back